src/main/java/Duke.java

```diff
@@ -1,10 +1,368 @@
+                }
+                else {
+                    System.out.println(Integer.toString(i+1) + ". " + tasks[i].getTaskType() + "[" + status +
+                            "] " + tasks[i].getDescription() + " (at: " + deadline + ")");
+                }
+            }
+        }
+    }
+
+    public static boolean qualifiesForPrint(String deadline) {
+        String[] deadline_split = deadline.split("-");
+        if (deadline_split.length == 3) {
+            return true; //qualifies
+        }
+        return false;
+    }
+
+    public static void markCompleted(int taskNum) {
+        System.out.println("Nice! I've marked this task as done:");
+        tasks[taskNum].updateisDone(true);
+        String deadline = tasks[taskNum].getDeadline();
+
+        if (tasks[taskNum].getTaskType().equals("[T]")) {
+            System.out.print("[T][" + tasks[taskNum].checkIfComplete() + "]" + tasks[taskNum].getDescription());
+        }
+        else if (tasks[taskNum].getTaskType().equals("[D]")) {
+            if (qualifiesForPrint(deadline)) {
+                System.out.print("[D][" + tasks[taskNum].checkIfComplete() + "]" + tasks[taskNum].getDescription() + " (by: ");
+                printDeadline(deadline);
+            } else {
+                System.out.print("[D][" + tasks[taskNum].checkIfComplete() + "]" + tasks[taskNum].getDescription() + " ( by: " + deadline + ")");
+            }
+        }
+        else {
+            if (qualifiesForPrint(deadline)) {
+                System.out.print("[E][" + tasks[taskNum].checkIfComplete() + "]" + tasks[taskNum].getDescription() + " (at: ");
+                printDeadline(deadline);
+            } else {
+                System.out.print("[E][" + tasks[taskNum].checkIfComplete() + "]" + tasks[taskNum].getDescription() + " ( by: " + deadline + ")");
+            }
+        }
+
+    }
+
+    public static void deleteTask(int taskNum) {
+        System.out.println("Noted. I've removed this task:");
+        if (tasks[taskNum].getTaskType().equals("[T]")) { // if it's a todo
+            System.out.println(tasks[taskNum].getTaskType() + tasks[taskNum].getCompletionStatus()
+                    + tasks[taskNum].getDescription());
+        }
+        else if (tasks[taskNum].getTaskType().equals("D")) { // it's a deadline
+            System.out.println(tasks[taskNum].getTaskType() + tasks[taskNum].getCompletionStatus()
+                    + tasks[taskNum].getDescription() + tasks[taskNum].getDeadline());
+        }
+        else { // it's an event liao
+            System.out.println(tasks[taskNum].getTaskType() + tasks[taskNum].getCompletionStatus()
+                    + tasks[taskNum].getDescription() + tasks[taskNum].getDeadline());
+        }
+
+        // here, we implement some for loop to shift all the tasks forward
+        for (int i = taskNum; i < counter-1; i++) {
+            tasks[i] = tasks[i+1];
+        }
+        counter--;
```

Comment:
You might want to use ArrayList as it removes the need for this.

Reply:
I second Yong Jie's suggestion, it would be more convenient to have ArrayList handle it for you.
However, if you prefer the [] operator offered by arrays, you can use System.arraycopy(), which is used by the remove() method in ArrayList.